### DISCUSSION TITLE
Privileged workflows using KubeVirt

### DISCUSSION THREAD

**Discussion by chriscarpenter12** [actions/actions-runner-controller#3297] [Ideas]
There's been a few issues/discussions opened about privileged workflows being ran and how to best isolate them. Some [suggesting Sysbox](https://github.com/actions/actions-runner-controller/discussions/977) using a new `RuntimeClassName`, but this doesn't really work for those running on a platform like OpenShift. Currently the two supported modes with ARC are `dind` and `k8s`. Without either being run with `privileged: true` it's not possible to take a workflow that runs in GitHub Actions cloud using `sudo` or `apt` for dependency installation and run it here. Having privileged enabled on the pods is a no-go in a lot of environments (including ours).

With the introduction of [Hooks](https://github.com/actions/runner-container-hooks) could a new mode be created to support running an isolated workflow using [KubeVirt](https://kubevirt.io/)?

I've done some minimal testing and it's possible to create a `VirtualMachineInstance` using an ephemeral qcow2 `containerdisk` created with Packer the same way the cloud [runner-images](https://github.com/actions/runner-images) are created for Azure.

I think it really comes down to coordination of the management of the jobs. Should the the runner pod that executes the hook be responsible for dispatching each step script similar to how it does in the `k8s` mode or could the `EphemeralRunner` spawn a `VirtualMachineInstance` directly that has the [runner](https://github.com/actions/runner) preinstalled?

With running an ephemeral VM it would also allow ARC to support Windows workloads as well.

**Comment by simonjohansson** [comment]
Being able to spawn `VirtualMachineInstance` would be amazing!
